Offices/Offices.Tests/DoctorServiceTests.cs

```diff
@@ -0,0 +1,314 @@
+        {
+            var configuration = new MapperConfiguration(cfg =>
+            {
+                cfg.AddProfile<ApplicationMappingProfile>();
+            });
+
+            _mapper = configuration.CreateMapper();
+        }
+
+        [Fact]
+        public async void GetDoctorById_Returns_Doctor()
+        {
+            // Arrange
+            var fixture = new Fixture();
+            var doctorRepositoryMock = new Mock<IDoctorRepository>();
+            doctorRepositoryMock.Setup(repo => repo.GetAsync(It.IsAny<string>(), CancellationToken.None))
+                              .ReturnsAsync(new Doctor
+                              {
+                                  Id = "1231231-1231-123",
+                                  FirstName = "John",
+                                  LastName = fixture.Create<string>(),
+                                  MiddleName = fixture.Create<string>(),
+                                  OfficeId = fixture.Create<string>(),
+                                  Status = fixture.Create<string>()
+                              });
+
+            var officeRepositoryMock = new Mock<IOfficeRepository>();
+
+            var doctorService = new DoctorService(doctorRepositoryMock.Object, officeRepositoryMock.Object, _mapper);
+
+            // Act
+            var doctorModel = await doctorService.Get("1231231-1231-123", CancellationToken.None);
+
+            // Assert
+            Assert.NotNull(doctorModel);
+            Assert.Equal("John", doctorModel.FirstName);
+        }
+
+        [Fact]
+        public async void CreateDoctor_ShouldBe_Success()
+        {
+            // Arrange
+            var fixture = new Fixture();
+            var officeRepositoryMock = new Mock<IOfficeRepository>();
+            officeRepositoryMock.Setup(repo => repo.GetAsync(It.IsAny<string>(), CancellationToken.None))
+                              .ReturnsAsync(new Office
+                              {
+                                  Id = "1231231-1231-123",
+                                  Address = fixture.Create<string>(),
+                                  PhotoURL = fixture.Create<string>(),
+                                  RegistryPhoneNumber = fixture.Create<string>(),
+                                  IsActive = true
+                              });
+
+            var doctorRepositoryMock = new Mock<IDoctorRepository>();
+
+            var doctorService = new DoctorService(doctorRepositoryMock.Object, officeRepositoryMock.Object, _mapper);
+
+            var createDoctorModel = new CreateDoctorModel
+            (
+                FirstName: fixture.Create<string>(),
+                LastName: fixture.Create<string>(),
+                MiddleName: fixture.Create<string>(),
+                OfficeId: "1231231-1231-123",
+                Status: fixture.Create<string>()
+            );
+
+            // Act
+            var exception = await Record.ExceptionAsync(async () =>
```

Comment:
Looks strange; any other ideas how to get exception from a delegate?

Reply:
Fixed, added shouldly